CONTRIBUTING: Add note about how to commit new maintainers-list.nix entry (#248013)
NixOS/nixpkgs: CONTRIBUTING.md

    The old config generation system used impure shell scripts and could break in specific circumstances (see #1234).

>   If you add yourself as maintainer, use `maintainers: add <handle>` as commit message ([see also](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers) and as a seperate commit.

Reviewer: ```suggestion
  In case of adding yourself as maintainer, do it in a separated commit (no need for a Pull Request), with message `maintainers: add <handle>`. See also ([the section of manual about meta.maintainers](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).
```

Reviewer 2: It isn't even necessarily adding yourself. It's adding anyone to the maintainer list.

Reviewer: This is the most common scenario: a newcomer adding a package.
But point taken, I will update this.

>   In case of adding yourself as maintainer, do it in a separated commit (no need for a Pull Request), with message `maintainers: add <handle>`. See also ([the section of manual about meta.maintainers](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).

Reviewer 3: ```suggestion
  When adding yourself as maintainer, in the same pull request make a separate commit with the message `maintainers: add <handle>`.
  See [Nixpkgs Maintainers](../maintainers/README.md) for details.
```

Given the information will be expanded in https://github.com/NixOS/nixpkgs/pull/250344, we should rather link to that file. We can still add information on NixOS contributions later.

    The old config generation system used impure shell scripts and could break in specific circumstances (see #1234).

>   When adding yourself as maintainer, in the same pull request make a separate commit with the message `maintainers: add <handle>`.

Reviewer 3: ```suggestion
  When adding yourself as maintainer, in the same pull request make a separate commit with the message `maintainers: add <handle>`.
Add the commit before those making changes to the package or module.
```

Reviewer 4: Is it intentional, or is the indentation off?

Reviewer 3: Ech, no, that was a typo.

This means that, when addressing review comments in order to keep the pull request in an always mergeable status, you will sometimes need to rewrite your branch's history and then force-push it with `git push --force-with-lease`.
Useful git commands that can help a lot with this are `git commit --patch --amend` and `git rebase --interactive`. For more details consult the git man pages or online resources like [git-rebase.io](https://git-rebase.io/) or [The Pro Git Book](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

> If you add yourself as maintainer, use `maintainers: add <handle>` as commit message ([see also](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).

Reviewer 2: Yes, I would put the commit message format in as an example and change the template accordingly under "Submitting Changes" and also try to make it more obvious that changes to `maintainer-list.nix` should occur in a separate commit.

I think it might be reasonable to also normalize `doc/<page>:` and `lib/<module>:` commit messages or simply `doc:` and `lib:` for more general changes, but that isn't necessary here.

> If you add yourself as maintainer, use `maintainers: add <handle>` as commit message ([see also](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).

Reviewer 5: Ah yeah then I guess this fits very well into the scope of the contributing documentation cleanup in #245243, because I create separate contributing documents for each part of Nixpkgs, such as `pkgs/README.md`, `doc/README.md`, `maintainers/README.md`, etc. And this would fit very well into `maintainers/README.md`.

> If you add yourself as maintainer, use `maintainers: add <handle>` as commit message ([see also](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).

Author: I added the change requested by @infinisil as a commit to this PR, which I can squash if requested of course.

Packages with automated tests are much more likely to be merged in a timely fashion because it doesn’t require as much manual testing by the maintainer to verify the functionality of the package. If there are existing tests for the package, they should be run to verify your changes do not break the tests. Tests can only be run on Linux. For more details on writing and running tests, see the [section in the NixOS manual](https://nixos.org/nixos/manual/index.html#sec-nixos-tests).

> When adding yourself as maintainer, in the same pull request make a separate commit with the message `maintainers: add <handle>`.

Reviewer 6: ```suggestion
When adding yourself as maintainer, in the same pull request, make a separate commit with the message `maintainers: add <handle>`.
```
